core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -77,13 +78,22 @@ public void build(AnalysisService analysisService) {
             tokenFilters.add(tokenFilter);
         }
 
-        int positionOffsetGap = analyzerSettings.getAsInt("position_offset_gap", 0);
-        int offsetGap = analyzerSettings.getAsInt("offset_gap", -1);
+        if (analyzerSettings.getAsMap().containsKey("position_increment_gap")
+                && analyzerSettings.getAsMap().containsKey("position_offset_gap")){
+            throw new IllegalArgumentException("Custom Analyzer [" + name() + "] has both [position_offset_gap] and [position_increment_gap]");
+        }
+
+        int positionIncrementGap = 0;
+        if (Version.indexCreated(indexSettings).before(Version.V_2_0_0) ){
```

Comment:
This change maintains support for the old name on 1.x indexes. If this were a common setting, I might agree that an extended deprecation period would be helpful. But this is an advanced setting, and especially with the change to the default, users should not be setting this often. Major releases are the time to break things, and I don't think we should maintain a very confusing setting name for all of 2.x.

We can however give a nicer error message in this case. So if we detect the setting used in 2.0+, we can throw an exception early (essentially a hard deprecation message), vs letting it fall through to have the setting be unknown. Something simple like "you must now switch to position_increment_gap".